src/software/ai/hl/stp/play/indirect_free_kick_play.cpp

```diff
@@ -0,0 +1,293 @@
+#include "software/ai/hl/stp/play/indirect_free_kick_play.h"
+
+#include <g3log/g3log.hpp>
+
+#include "shared/constants.h"
+#include "software/ai/hl/stp/evaluation/ball.h"
+#include "software/ai/hl/stp/evaluation/possession.h"
+#include "software/ai/hl/stp/play/corner_kick_play.h"
+#include "software/ai/hl/stp/play/play_factory.h"
+#include "software/ai/hl/stp/tactic/chip_tactic.h"
+#include "software/ai/hl/stp/tactic/crease_defender_tactic.h"
+#include "software/ai/hl/stp/tactic/goalie_tactic.h"
+#include "software/ai/hl/stp/tactic/move_tactic.h"
+#include "software/ai/hl/stp/tactic/passer_tactic.h"
+#include "software/ai/hl/stp/tactic/receiver_tactic.h"
+#include "software/ai/hl/stp/tactic/shoot_goal_tactic.h"
+#include "software/ai/passing/pass_generator.h"
+#include "software/util/logger/custom_logging_levels.h"
+
+using namespace Passing;
+
+const std::string IndirectFreeKickPlay::name = "Indirect Free Kick Play";
+
+IndirectFreeKickPlay::IndirectFreeKickPlay()
+    : MAX_TIME_TO_COMMIT_TO_PASS(Duration::fromSeconds(3))
+{
+}
+
+std::string IndirectFreeKickPlay::getName() const
+{
+    return IndirectFreeKickPlay::name;
+}
+
+bool IndirectFreeKickPlay::isApplicable(const World &world) const
+{
+    double min_dist_to_corner =
+        std::min((world.field().enemyCornerPos() - world.ball().position()).len(),
+                 (world.field().enemyCornerNeg() - world.ball().position()).len());
+
+    // Make sure we don't interfere with the cornerkick play
+    return world.gameState().isOurFreeKick() &&
+           min_dist_to_corner >= CornerKickPlay::BALL_IN_CORNER_RADIUS;
```

Comment:
Can we do this as part of refactoring? That seems out of scope for this PR, but not a bad thing to revisit. Added to #968

Reply:
Actually I don't think they should be made static because then we lose the benefit of inheritance / enforced interfaces, but that's a discussion for another time.